src/Components/Aspire.Azure.Messaging.EventHubs/EventHubsComponent.cs

```diff
@@ -56,15 +55,15 @@ protected static string GetNamespaceFromSettings(AzureMessagingEventHubsSettings
                 : new Uri(settings.FullyQualifiedNamespace).Host;
 
             // This is likely to be similar to {yournamespace}.servicebus.windows.net or {yournamespace}.servicebus.chinacloudapi.cn
-            if (ns.Contains(".servicebus", StringComparison.OrdinalIgnoreCase))
+            var serviceBusIndex = ns.IndexOf(".servicebus", StringComparison.OrdinalIgnoreCase);
+            if (serviceBusIndex != -1)
             {
-                ns = ns[..ns.IndexOf(".servicebus")];
+                ns = ns[..serviceBusIndex];
             }
             else
             {
-                // Use a random prefix if no meaningful name is found e.g., "localhost", "127.0.0.1".
-                // This is used to create blob containers names that are unique in the referenced storage account.
-                RandomNumberGenerator.GetHexString(12, true);
+                // sanitize the namespace if it's not a servicebus namespace
+                ns = ns.Replace(".", "-");
```

Comment:
Was wondering if it be based on an allow-list, assuming we only expect but this will be a domain here so in theory this should be fine.

Reply:
Huh. How did that get orphaned? Am I tagged because git blame tattled on me?

Reply:
> How did that get orphaned?

Just a simple mistake (not yours 😉).

> Am I tagged because git blame tattled on me?

I just tagged as an FYI since you've been heavily involved in the EventHubs code.